server/src/main/java/org/elasticsearch/index/query/NestedQueryBuilder.java

```diff
@@ -317,10 +318,14 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
     @Override
     public void extractInnerHitBuilders(Map<String, InnerHitContextBuilder> innerHits) {
         if (innerHitBuilder != null) {
+            if (innerHits.containsKey(innerHitBuilder.getName())) {
```

Comment:
here the `path` is used if the name is null, can you infer the name first and then check inside the map ?